keras_cv/models/segmentation/fcn.py

```diff
@@ -0,0 +1,677 @@
+            from the data should be represented by integers with range
+            [0, classes).
+        backbone: a backbone network for the model. Can be a `tf.keras.models.Model`
+            instance. The supported pre-defined backbone models are:
+            1. "vgg16", a VGG16 model
+            2. "vgg19", a VGG19 model
+            Defaults to 'vgg16'.
+            Note: If a custom `tf.keras.models.Model` is passed, then only the
+            `Conv2D`, `MaxPooling2D` and `Dense` layers are extracted from it
+            to make the custom backbone.
+        model_architecture: str, defines the model architecture based on the
+            implementation details present in the paper. The supported
+            architectures are:
+            1. 'fcn8s', a FCN-8S definition
+            2. 'fcn16s', a FCN-16S definition
+            3. 'fcn32s', a FCN-32S definition
+            Defaults to 'fcn8s'.
+        input_shape: `list` or `tuple`. Defines the shape of the tensor to be expected as input.
+        include_rescaling: bool, one of True or False. Defines whether to use a Rescaling layer or not.
+        return_mask: bool, one of True or False. Returns a 1-channel result instead of a
+            num_classes-channel result.
+        return_dtype: `tf.dtypes`. Return results in the chosen dtype. Defaults to tf.float32
+        include_imagenet_weights: bool, one of True or False. Defines whether to use
+            original ImageNet weights for VGG backbones.
+    """
+
+    def __init__(
+        self,
+        classes,
+        backbone,
+        model_architecture=None,
+        input_shape=(224, 224, 3),
+        include_rescaling=False,
+        return_mask=False,
+        return_dtype=tf.float32,
+        include_imagenet_weights=False,
+    ):
+
+        if isinstance(backbone, tf.keras.models.Model):
+            if model_architecture is not None:
+                raise ValueError(
+                    "`model_architecture` cannot be set if `backbone` is not in ['vgg16', 'vgg19']. Either set `backbone` to one of the accepted values or remove the `model_architecture` argument."
+                )
+            if include_imagenet_weights:
+                raise ValueError(
+                    """`include_imagenet_weights` cannot be set if `backbone` is not in ['vgg16', 'vgg19']. Either set `backbone` to one of the accepted values or remove the
+                    `include_imagenet_weights` argument."""
+                )
+            else:
+                backbone = CustomArchitectureBuilder(backbone)
+                classes_conv = tf.keras.layers.Conv2D(
+                    filters=classes,
+                    kernel_size=(1, 1),
+                    strides=(1, 1),
+                    padding="same",
+                    activation="softmax",
+                )
+
+                input_tensor = tf.keras.Input(input_shape)
+                if include_rescaling:
```

Comment:
The function that will parse all layers only accepts Conv2D, MaxPooling and Dense layers. So the Rescaling layer will have to stay here itself.

Reply:
Segmentation models shouldn't take the `input_rescaling` argument. The input to the semantic segmentation model is passed through the backbone which might or might not rescale the input. The downstream model is agnostic to this.

If a backbone cannot rescale (because we're filtering layers), the API for segmentation models shouldn't change